src/internal/FluencePeer.ts

```diff
@@ -0,0 +1,350 @@
+    isConnected: Boolean;
+
+    /**
+     * The Peer's identification in the Fluence network
+     */
+    selfPeerId: PeerIdB58;
+
+    /**
+     * The list of relays's peer ids to which the peer is connected to
+     */
+    connectedRelays: Array<PeerIdB58>;
+}
+
+/**
+ * This class implements the Fluence protocol for javascript-based environments.
+ * It provides all the necessary features to communicate with Fluence network
+ */
+export class FluencePeer {
+    // TODO:: implement api alongside with multi-relay implementation
+    //async addConnection(relays: Array<ConnectionSpec>): Promise<void> {}
+
+    // TODO:: implement api alongside with multi-relay implementation
+    //async removeConnections(relays: Array<ConnectionSpec>): Promise<void> {}
+
+    /**
+     * Creates a new Fluence Peer instance. Does not start the workflows.
+     * In order to work with the Peer it has to be initialized with the `init` method
+     */
+    constructor() {}
+
+    /**
+     * Get the information about Fluence Peer connections
+     */
+    get connectionInfo(): ConnectionInfo {
+        const isConnected = this._connection?.isConnected();
+        return {
+            isConnected: isConnected,
+            selfPeerId: this._selfPeerId,
+            connectedRelays: this._relayPeerId ? [this._relayPeerId] : [],
+        };
+    }
+
+    /**
+     * Initializes the peer: starts the Aqua VM, initializes the default call service handlers
+     * and (optionally) connect to the Fluence network
+     * @param config - object specifying peer configuration
+     */
+    async init(config?: PeerConfig): Promise<void> {
+        let peerId;
+        const peerIdOrSeed = config?.peerId;
+        if (!peerIdOrSeed) {
+            peerId = await randomPeerId();
+        } else if (isPeerId(peerIdOrSeed)) {
+            // keep unchanged
+            peerId = peerIdOrSeed;
+        } else {
+            // peerId is string, therefore seed
+            peerId = await peerIdFromEd25519SK(peerIdOrSeed);
+        }
+        this._selfPeerIdFull = peerId;
+
+        await this._initAirInterpreter(config?.avmLogLevel || 'off');
+
+        this._callServiceHandler = makeDefaultClientHandler();
+
+        if (config?.connectTo) {
+            let connectTo;
+            if (Array.isArray(config!.connectTo)) {
+                connectTo = config!.connectTo;
+            } else {
+                connectTo = [config!.connectTo];
+            }
+
+            let theAddress: ConnectionSpec;
+            let fromNode = (connectTo[0] as any).multiaddr;
+            if (fromNode) {
+                theAddress = new MA(fromNode);
+            } else {
+                theAddress = new MA(connectTo[0] as string);
+            }
+
+            await this._connect(theAddress);
+        }
+    }
+
+    /**
+     * Uninitializes the peer: stops all the underltying workflows, stops the Aqua VM
+     * and disconnects from the Fluence network
+     */
+    async uninit() {
```

Comment:
`uninit` => `stop`

Reply:
Technically it does more than just stopping the avm, e.g disconnecting from the network if there is any open connection.
Plus there is the symmetry between `init`,  `uninit` functions